ambry-messageformat/src/test/java/com.github.ambry.messageformat/MessageSievingInputStreamTest.java

```diff
@@ -560,109 +653,348 @@ private void testDeletedBlobs(short blobVersion, BlobType blobType) throws IOExc
+    Assert.assertArrayEquals(key.toBytes(), keyInStreamBytes);
+    Assert.assertTrue(header.isPutRecord());
+    ByteBuffer encryptionKeyInStream =
+        header.hasEncryptionKeyRecord() ? deserializeBlobEncryptionKey(sievedStream) : null;
+    BlobProperties propsFromStream = deserializeBlobProperties(sievedStream);
+    ByteBuffer userMetadataFromStream = deserializeUserMetadata(sievedStream);
+    BlobData blobDataFromStream = deserializeBlob(sievedStream);
+    Assert.assertEquals(encryptionKey == null, encryptionKeyInStream == null);
+    if (encryptionKey != null) {
+      Assert.assertArrayEquals(encryptionKey, encryptionKeyInStream.array());
+    }
+    Assert.assertEquals(serviceId, propsFromStream.getServiceId());
+    Assert.assertEquals(accountId, propsFromStream.getAccountId());
+    Assert.assertEquals(containerId, propsFromStream.getContainerId());
+    Assert.assertEquals(ByteBuffer.wrap(usermetadata), userMetadataFromStream);
+    Assert.assertEquals(ByteBuffer.wrap(data), blobDataFromStream.getStream().getByteBuffer());
+    Assert.assertEquals(blobType, blobDataFromStream.getBlobType());
+  }
+}
+
+/**
+ * An implementation of the {@link StoreKeyConverter} interface that generates random key mappings for the given set of
+ * keys. The mappings are deterministic (a key will always be mapped to the same converted keys). The converted keys can
+ * be of length less than, equal to, or greater than the size of the input key. Additionally, a set of "invalid" keys
+ * can be provided while instantiating, and for these keys a null mapping will be provided.
+ */
+class RandomKeyConverter implements StoreKeyConverter {
+
+  Collection<? extends StoreKey> invalids = Collections.emptyList();
+  Map<StoreKey, StoreKey> onceConverted = new HashMap<>();
+
+  /**
+   * Add invalid mappings.
+   * @param invalids the keys for which no mapping will be generated during conversion.
+   */
+  public void addInvalids(Collection<? extends StoreKey> invalids) {
+    this.invalids = invalids;
+  }
+
+  @Override
+  public Map<StoreKey, StoreKey> convert(Collection<? extends StoreKey> input) {
+    Map<StoreKey, StoreKey> output = new HashMap<>();
+    input.forEach(inKey -> {
+      if (onceConverted.containsKey(inKey)) {
+        output.put(inKey, onceConverted.get(inKey));
+      } else {
+        StoreKey replaceMent = invalids.contains(inKey) ? null : new MockId(
+            inKey.getID().substring(0, inKey.getID().length() / 2) + Integer.toString(RANDOM.nextInt(1000)));
+        onceConverted.put(inKey, replaceMent);
+        output.put(inKey, replaceMent);
+      }
+    });
+    return output;
+  }
+
+  @Override
+  public StoreKey getConverted(StoreKey storeKey) {
+    return onceConverted.get(storeKey);
+  }
+}
+
+/**
+ * A {@link Transformer} implementation that does random key conversions, used for testing.
+ */
+class ValidatingKeyConvertingTransformer implements Transformer {
```

Comment:
did not look at this closely (consider using `BlobIdTransformer` or does that not work because you deal with `MockId` and that deals with `BlobId`?)

Reply:
I will look into this in a subsequent patch (there is work involved to make changes to `MockStoreKeyConverterFactory` that I will be working on).